transport/grpc/outbound.go

```diff
@@ -42,16 +45,24 @@ func (outbound) Options() (o transport.Options) {
 }
 
 func (o outbound) Call(ctx context.Context, req *transport.Request) (*transport.Response, error) {
+	start := time.Now()
+	deadline, _ := ctx.Deadline()
+	ttl := deadline.Sub(start)
+
 	requestBody, err := ioutil.ReadAll(req.Body)
 	if err != nil {
 		return nil, err
 	}
 
 	metadataHeaders := getRequestHeaders(ctx, req)
 	ctx = metadata.NewContext(ctx, metadataHeaders)
-
 	uri := fmt.Sprintf("/%s/%s", url.QueryEscape(req.Service), url.QueryEscape(req.Procedure))
-	return callDownstream(ctx, uri, &requestBody, o.conn)
+
+	response, err := callDownstream(ctx, uri, &requestBody, o.conn)
+	if err != nil {
+		return nil, getErrFromGRPCError(err, req, ttl)
```

Comment:
you can make `getErrFromGRPCError` return `nil` if `err` is nil and then this
simplifies to,

```
return response, getErrFromGRPCError(err, req, ttl)
```